CONTRIBUTING.adoc

```diff
@@ -0,0 +1,59 @@
+= tbtc Contribution Guide
+
+👍🎉 First off, thanks for taking the time to contribute! 🎉👍 Contributions
+are welcome from anyone on the internet, and even the smallest of fixes
+are appreciated!
+
+The following is a set of guidelines for contributing to tBTC and its
+packages. These are mostly guidelines, not rules. Use your best
+judgment, and feel free to propose changes to this document in a pull
+request.
+
+== Getting started
+
+1.  Fork https://github.com/keep-network/tbtc[`keep-network/tbtc`]
+2.  Clone your fork
+3.  Follow the
+https://github.com/keep-network/tbtc/tree/master/implementation#setup[installation
+& build steps] in the repo's top-level README.
+4.  Setup the recommended link:#development-tooling[Development
+Tooling].
+5.  Open a PR against the `master` branch and describe the change you
+are intending to undertake in the PR description.
+
+Before marking the PR as open for review, make sure:
+
+* It passes the linter checks (`npm run lint`)
+* It passes the
+https://app.circleci.com/github/keep-network/tbtc/pipelines[continuous
+integration tests] on CircleCI.
+* Your changes have sufficient test coverage (e.g regression tests have
+been added for bug fixes, unit tests for new features)
+
+== Development Tooling
+
+Commits
+https://help.github.com/en/articles/about-commit-signature-verification[must
+be signed].
+
+=== Continuous Integration
+
+We use https://circleci.com[CircleCI] for continuous integration. All
+CircleCI jobs (including tests, linting) must be green to merge a PR.
+
+=== Pre-commit
+
+Pre-commit is a tool to install hooks that check code before commits are
+made. It can be helpful to install this, to automatically run linter
+checks. Follow the https://pre-commit.com/[installation instructions
+here], and then run `pre-commit install` to install the hooks.
+
+=== Linting
+
+Linters for Solidity and JavaScript code are setup and run automatically
+as part of pre-commit hooks.
+
+If you want to change a rule, or add a custom rule, please make these
```

Comment:
please propose*

The chances of linter adjustments landing are probably 50-50 at best :wink: